sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -202,6 +202,9 @@ def __init__(self, fully_qualified_namespace, eventhub_name, credential, **kwarg
         self._container_id = CONTAINER_PREFIX + str(uuid.uuid4())[:8]
         if isinstance(credential, AzureSasCredential):
             self._credential = AzureSasTokenCredential(credential)
+        elif isinstance(credential, AzureNamedKeyCredential):
+            name, key = credential.named_key
+            self._credential = EventHubSharedKeyCredential(name, key) # type: ignore
```

Comment:
shouldn't we pass the credential object and extract value the same way as the `AzureSasCredential` one so that the name and key could be rotated?
otherwise if users call `AzureNamedKeyCredential.udpate(name, key)`, that doesn't change the value in `EventHubSharedKeyCredential`.